src/improvements/template/TransferL2PAOFromL1.sol

```diff
@@ -0,0 +1,95 @@
+
+    /// @notice Returns the safe address string identifier
+    function safeAddressString() public pure override returns (string memory) {
+        return "ProxyAdminOwner";
+    }
+
+    /// @notice Returns the storage write permissions required for this task.
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](1);
+        storageWrites[0] = "OptimismPortalProxy";
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with the new owner from a TOML file.
+    function _templateSetup(string memory taskConfigFilePath) internal override {
+        super._templateSetup(taskConfigFilePath);
+        string memory toml = vm.readFile(taskConfigFilePath);
+
+        // New owner address. This address is unaliased.
+        address newOwnerToAlias = abi.decode(vm.parseToml(toml, ".newOwnerToAlias"), (address));
+        // Apply the alias to the new owner.
+        aliasedNewOwner = AddressAliasHelper.applyL1ToL2Alias(newOwnerToAlias);
+
+        // Only allow one chain to be modified at a time with this template.
+        SuperchainAddressRegistry.ChainInfo[] memory _chains =
+            abi.decode(vm.parseToml(toml, ".l2chains"), (SuperchainAddressRegistry.ChainInfo[]));
+        require(_chains.length == 1, "Must specify exactly one chain id to transfer ownership for");
+    }
+
+    /// @notice Builds the actions for transferring ownership of the proxy admin on the L2.
+    function _build() internal override {
+        uint64 gasLimit = 200000; // This gas limit was used for an example task previously: tasks/sep/010-op-l2-predeploy-upgrade-from-l1/input.json
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        OptimismPortal optimismPortal =
+            OptimismPortal(superchainAddrRegistry.getAddress("OptimismPortalProxy", chains[0].chainId));
+        optimismPortal.depositTransaction(
+            address(Predeploys.PROXY_ADMIN),
+            0,
+            gasLimit,
+            false,
+            abi.encodeWithSelector(ProxyAdmin.transferOwnership.selector, aliasedNewOwner)
+        );
+    }
+
+    /// @notice Validates that the owner was transferred correctly.
+    function _validate(VmSafe.AccountAccess[] memory, Action[] memory) internal view override {
+        // TODO: We can't currently perform an assertion on the L2 because the transaction is only simulated and not actually executed.
+        // assertEq(ProxyAdmin(l2ProxyAdminPredeploy).owner(), aliasedNewOwner, "aliased new owner not set correctly");
+    }
+
+    /// @notice Aliased new owner is a code exception. This is because the aliased address is not a contract.
+    function getCodeExceptions() internal view virtual override returns (address[] memory) {
+        address[] memory codeExceptions = new address[](1);
+        codeExceptions[0] = aliasedNewOwner;
+        return codeExceptions;
```

Comment:
I don't think we need code exceptions since this storage write happens on L2

Reply:
done https://github.com/ethereum-optimism/superchain-ops/pull/969/commits/8dc540942082539282624e72c2cc046f9f977601